Bricks/strings/util.h

```diff
@@ -25,33 +25,56 @@ SOFTWARE.
 #ifndef BRICKS_STRINGS_UTIL_H
 #define BRICKS_STRINGS_UTIL_H
 
-#include <cstddef>
-#include <type_traits>
-#include <cstring>
-#include <cctype>
 #include <algorithm>
-#include <iterator>
+#include <cstring>
+#include <sstream>
+#include <string>
 
-// These are small headers, and #include "bricks/strings/util.h" should be enough.
-#include "printf.h"
-#include "fixed_size_serializer.h"
-#include "join.h"
-#include "split.h"
-#include "distance.h"
-#include "chunk.h"
+#include "is_string_type.h"
+
+#include "../template/decay.h"
 
 namespace bricks {
 namespace strings {
 
+template <typename DECAYED_T>
+struct ToStringImpl {
+  template <typename T>
+  static std::string ToString(T&& something) {
+    static_assert(std::is_same<bricks::decay<T>, DECAYED_T>::value, "");
+    return std::to_string(std::forward<T>(something));
+  }
+};
+
+template <>
+struct ToStringImpl<std::string> {
+  template <typename T>
+  static std::string ToString(T&& string) {
+    return string;
+  }
+};
+
+template <>
+struct ToStringImpl<char*> {  // Decayed type in template parameters list.
+  static std::string ToString(const char* string) { return string; }
+};
+
+template <int N>
+struct ToStringImpl<char[N]> {  // Decayed type in template parameters list.
+  static std::string ToString(const char chunk[N]) {
```

Comment:
Maybe rename `chunk` to `string`? IMHO it's not clear from the name that `chunk`must be null-terminated.